bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/editor/EditorConfigurationPage.java

```diff
@@ -0,0 +1,259 @@
+		composite.setFont(parent.getFont());
+		control = createPreferenceContent(composite, isProjectScope);
+		control.setLayoutData(new GridData(GridData.FILL_BOTH));
+		if (isProjectScope) {
+			enableProjectSpecificSettings(hasProjectSpecificOptions());
+		}
+		refreshWidgets(configuration.options(getElement()));
+		Dialog.applyDialogFont(composite);
+		return composite;
+	}
+
+	protected ConfigurationArea getConfigurationArea(Composite composite, boolean isProjectScope) {
+		return new EditorConfigurationArea(composite, (EditorMetadata) configuration.metadata(), isProjectScope);
+	}
+
+	private Control createPreferenceContent(Composite parent, boolean isProjectScope) {
+		Composite composite = new Composite(parent, SWT.NONE);
+		composite.setLayout(GridLayoutFactory.fillDefaults().create());
+		composite.setFont(parent.getFont());
+		area = getConfigurationArea(composite, isProjectScope);
+		return composite;
+	}
+
+	protected void refreshWidgets(Object options) {
+		setErrorMessage(null);
+		area.load(options, useProjectSettings() || !projectScope().isPresent());
+	}
+
+	protected Optional<ProjectScope> projectScope() {
+		return new ResolveProjectScope(workspace).apply(getElement());
+	}
+
+	@Override
+	protected void performDefaults() {
+		if (useProjectSettings()) {
+			enableProjectSpecificSettings(false);
+		}
+		IEclipsePreferences prefs = manager.getWorkingCopy(scope().getNode(configuration.qualifier()));
+		try {
+			for (String key : prefs.keys()) {
+				prefs.remove(key);
+			}
+		} catch (BackingStoreException e) {
+			Platform.getLog(getClass()).error("Unable to restore default values.", e); //$NON-NLS-1$
+		}
+		refreshWidgets(configuration.defaults());
+		super.performDefaults();
+	}
+
+	@Override
+	public boolean performOk() {
+		IEclipsePreferences prefs;
+		if (projectScope().isPresent()) {
+			prefs = manager.getWorkingCopy(projectScope().get().getNode(configuration.qualifier()));
+			if (!useProjectSettings()) {
+				try {
+					for (String key : prefs.keys()) {
+						prefs.remove(key);
+					}
+				} catch (BackingStoreException e) {
+					Platform.getLog(getClass()).error("Unable to reset project preferences.", e); //$NON-NLS-1$
+				}
+				prefs = null;
+			}
+		} else {
+			prefs = manager.getWorkingCopy(InstanceScope.INSTANCE.getNode(configuration.qualifier()));
+		}
+		if (prefs != null) {
+			area.store(prefs);
+		}
+		try {
+			manager.applyChanges();
+		} catch (BackingStoreException e) {
+			Platform.getLog(getClass()).error("Unable to save preferences.", e); //$NON-NLS-1$
+			return false;
+		}
+		return true;
+	}
+
+	private IScopeContext scope() {
+		return projectScope().map(IScopeContext.class::cast).orElse(InstanceScope.INSTANCE);
+	}
+
+	protected boolean hasProjectSpecificOptions() {
+		return projectScope()//
+				.map(p -> p.getNode(configuration.qualifier()))//
+				.map(n -> n.get(((EditorMetadata) configuration.metadata()).preferLspEditor().identifer(), null))//
```

Comment:
I think this line is the source of the not restoring the project-specific checkbox properly.

Reply:
Should be fixed now